Mathlib/LinearAlgebra/QuadraticForm/Basic.lean

```diff
@@ -1291,37 +1328,37 @@ variable (R)
 
 The weights are applied using `•`; typically this definition is used either with `S = R` or
 `[Algebra S R]`, although this is stated more generally. -/
-def weightedSumSquares [Monoid S] [DistribMulAction S R] [SMulCommClass S R R] (w : ι → S) :
-    QuadraticForm R (ι → R) :=
-  ∑ i : ι, w i • proj i i
-#align quadratic_form.weighted_sum_squares QuadraticForm.weightedSumSquares
+def weightedSumSquares [Monoid S] [DistribMulAction S R] [SMulCommClass S R R] [SMulCommClass R S R]
+    (w : ι → S) : QuadraticMap R (ι → R) R :=
+  ∑ i : ι, w i • (proj (R:=R) (n:=ι) i i)
+#align quadratic_form.weighted_sum_squares QuadraticMap.weightedSumSquares
 
 end
 
 @[simp]
 theorem weightedSumSquares_apply [Monoid S] [DistribMulAction S R] [SMulCommClass S R R]
-    (w : ι → S) (v : ι → R) :
+    [SMulCommClass R S R] (w : ι → S) (v : ι → R) :
     weightedSumSquares R w v = ∑ i : ι, w i • (v i * v i) :=
-  QuadraticForm.sum_apply _ _ _
-#align quadratic_form.weighted_sum_squares_apply QuadraticForm.weightedSumSquares_apply
+  QuadraticMap.sum_apply _ _ _
+#align quadratic_form.weighted_sum_squares_apply QuadraticMap.weightedSumSquares_apply
 
 /-- On an orthogonal basis, the basis representation of `Q` is just a sum of squares. -/
 theorem basisRepr_eq_of_iIsOrtho {R M} [CommRing R] [AddCommGroup M] [Module R M]
-    [Invertible (2 : R)] (Q : QuadraticForm R M) (v : Basis ι R M)
+    [Invertible (2 : R)] (Q : QuadraticMap R M R) (v : Basis ι R M)
     (hv₂ : (associated (R := R) Q).IsOrthoᵢ v) :
     Q.basisRepr v = weightedSumSquares _ fun i => Q (v i) := by
   ext w
   rw [basisRepr_apply, ← @associated_eq_self_apply R, map_sum, weightedSumSquares_apply]
   refine' sum_congr rfl fun j hj => _
   rw [← @associated_eq_self_apply R, LinearMap.map_sum₂, sum_eq_single_of_mem j hj]
   · rw [LinearMap.map_smul, LinearMap.map_smul₂, smul_eq_mul, associated_apply, smul_eq_mul,
-      smul_eq_mul]
-    ring
+      smul_eq_mul, smul_eq_mul]
+    ring_nf
   · intro i _ hij
     rw [LinearMap.map_smul, LinearMap.map_smul₂,
       show associatedHom R Q (v i) (v j) = 0 from hv₂ hij, smul_eq_mul, smul_eq_mul,
       mul_zero, mul_zero]
 set_option linter.uppercaseLean3 false in
-#align quadratic_form.basis_repr_eq_of_is_Ortho QuadraticForm.basisRepr_eq_of_iIsOrtho
+#align quadratic_form.basis_repr_eq_of_is_Ortho QuadraticMap.basisRepr_eq_of_iIsOrtho
 
```

Comment:
One final time:
- This section should be in the namespace QuadraticForm instead of QuadraticMap.
- Every occurrence of QuadraticMap in this section should be replaced with QuadraticForm.